### DISCUSSION TITLE
testing

### DISCUSSION THREAD

**Discussion by berlinger-mschwartz** [laravel/framework#39533] [Q&A]
Hi everyone,

Hope you can help me out with something seemingly trivial.

I'm trying to configure a MySQL test database for my unit tests, because an in-memory SQLite database has proven to be too limited. I was hoping to use an `.env.testing` file for this.

Set-up:

* `.env` contains `DB_DATABASE=my_database`
* `.env.testing` is a copy of `.env`, but contains `DB_DATABASE=my_database_test`

Now if I run `php artisan config:clear` and `php artisan config:cache --env=testing`, the resulting config file `bootstrap/cache/config.php` contains `my_database` instead of `my_database_test`. The addition of the `--env` argument doesn't seem to matter.

What am I missing?

My current hypothesis is that our Docker set-up is somehow responsible for these problems. If I change `DB_DATABASE` in my `.env` and rerun the two config commands, the database is not updated until I recreate the container.

Any suggestions?

**Comment by jdecode** [comment]
The solution that I use (recommended solution, totally from personal experience, might differ from other popular answers).

Set DB URL in `.env.example` (and copy this file at the time of initial setup to `.env`) as :
`DATABASE_URL="pgsql://postgres:docker@docker-db-service:5432/postgres"`

Set DB URL in `.env.testing` as :
`DATABASE_URL="pgsql://postgres:docker@docker-db-service:5432/test_db"`

Set `DB_CONNECTION=pgsql` in docker-compose (since the connection does not need to change when switching from primary app to testing).

I have used `pgsql`, and the same principles apply to `mysql` as well.

Honestly, I am not sure if this is helpful, so feel free to let me know so that I can try to explain it better next time.
